tests/std/tests/P2093R14_formatted_output/test.cpp

```diff
@@ -0,0 +1,516 @@
+    test_sequence_closure("\xC3\xB1");
+
+    // Invalid 2 Octet Sequence
+    test_sequence_closure("\xC3\x28");
+
+    // Invalid Sequence Identifier
+    test_sequence_closure("\xA0\xA1");
+
+    // Valid 3 Octet Sequence
+    test_sequence_closure("\xE2\x82\xA1");
+
+    // Invalid 3 Octet Sequence (in 2nd Octet)
+    test_sequence_closure("\xE2\x28\xA1");
+
+    // Invalid 3 Octet Sequence (in 3rd Octet)
+    test_sequence_closure("\xE2\x82\x28");
+
+    // Valid 4 Octet Sequence
+    test_sequence_closure("\xF0\x90\x8C\xBC");
+
+    // Invalid 4 Octet Sequence (in 2nd Octet)
+    test_sequence_closure("\xF0\x28\x8C\xBC");
+
+    // Invalid 4 Octet Sequence (in 3rd Octet)
+    test_sequence_closure("\xF0\x90\x28\xBC");
+
+    // Invalid 4 Octet Sequence (in 4th Octet)
+    test_sequence_closure("\xF0\x28\x8C\x25");
+
+    fclose(temp_file_stream);
+}
+
+void test_stream_flush_console() {
+    // If the ordinary literal encoding is UTF-8, then the FILE stream associated with
+    // a console should always be flushed before writing output during a call to std::print()
+    // and std::println(). Otherwise, the stream should *NOT* be flushed.
+    test::win_console temp_console{};
+    FILE* const console_file_stream = temp_console.get_file_stream();
+
+    print(console_file_stream, "Hello,");
+
+    {
+        const wstring extractedStr{temp_console.get_console_line(0)};
+
+        if constexpr (_Is_ordinary_literal_encoding_utf8()) {
+            assert(extractedStr == L"Hello,");
+        } else {
+            assert(extractedStr.empty());
+        }
+    }
+
+    println(console_file_stream, " world!");
+
+    {
+        const wstring extractedStr{temp_console.get_console_line(0)};
+
+        if constexpr (_Is_ordinary_literal_encoding_utf8()) {
+            assert(extractedStr == L"Hello, world!");
+        } else {
+            assert(extractedStr.empty());
+        }
+    }
+
+    maybe_flush_console_file_stream(temp_console);
+
+    {
+        const wstring extractedStr{temp_console.get_console_line(0)};
+        assert(extractedStr == L"Hello, world!");
+    }
+}
+
+void test_stream_flush_file() {
+    // Regardless of the ordinary literal encoding, neither std::print() nor std::println()
+    // should flush file streams which do not refer to consoles.
+    string temp_file_name_str;
+
+    {
+        ofstream output_file_stream;
+
+        while (!output_file_stream.is_open()) {
+            temp_file_name_str.resize_and_overwrite(L_tmpnam_s, [](char* const dest_ptr, const size_t allocated_size) {
+                const errno_t tmpnam_result = tmpnam_s(dest_ptr, allocated_size);
+                assert(tmpnam_result == 0);
+
+                return allocated_size;
+            });
+
+            output_file_stream = ofstream{temp_file_name_str, ios::out};
+        }
+
+        print(output_file_stream, "Hello, ");
+
+        {
+            ifstream input_file_stream = ifstream{temp_file_name_str, ios::in};

```

Comment:
The temporary here is unnecessary; the variable can be directly initialized. Occurs below.